src/quat.js

```diff
@@ -447,36 +447,50 @@ export function fromMat3(out, m) {
 }
 
 /**
- * Creates a quaternion from the given euler angle x, y, z.
+ * Creates a quaternion from the given euler angle z, y, x.
  *
  * @param {quat} out the receiving quaternion
- * @param {x} Angle to rotate around X axis in degrees.
- * @param {y} Angle to rotate around Y axis in degrees.
  * @param {z} Angle to rotate around Z axis in degrees.
+ * @param {y} Angle to rotate around Y axis in degrees.
+ * @param {x} Angle to rotate around X axis in degrees.
  * @returns {quat} out
  * @function
  */
-export function fromEuler(out, x, y, z) {
+export function fromEuler(out, z, y, x) {
   let halfToRad = (0.5 * Math.PI) / 180.0;
-  x *= halfToRad;
-  y *= halfToRad;
   z *= halfToRad;
+  y *= halfToRad;
+  x *= halfToRad;
 
-  let sx = Math.sin(x);
-  let cx = Math.cos(x);
-  let sy = Math.sin(y);
-  let cy = Math.cos(y);
   let sz = Math.sin(z);
   let cz = Math.cos(z);
+  let sy = Math.sin(y);
+  let cy = Math.cos(y);
+  let sx = Math.sin(x);
+  let cx = Math.cos(x);
 
-  out[0] = sx * cy * cz - cx * sy * sz;
-  out[1] = cx * sy * cz + sx * cy * sz;
-  out[2] = cx * cy * sz - sx * sy * cz;
-  out[3] = cx * cy * cz + sx * sy * sz;
+  out[0] = sz * cy * cx - cz * sy * sx;
+  out[1] = cz * sy * cx + sz * cy * sx;
+  out[2] = cz * cy * sx - sz * sy * cx;
+  out[3] = cz * cy * cx + sz * sy * sx;
 
   return out;
 }
 
+/**
+ * Creates a quaternion from the given euler angle x, y, z.
+ *
+ * @param {quat} out the receiving quaternion
+ * @param {x} Angle to rotate around X axis in degrees.
+ * @param {y} Angle to rotate around Y axis in degrees.
+ * @param {z} Angle to rotate around Z axis in degrees.
+ * @returns {quat} out
+ * @function
+ */
+export function fromEulerXYZ(out, x, y, z) {
+  return fromEuler(out, z, y, x);
+}
```

Comment:
Just swapping the order in which angles are passed into the function does not change the rotation order -- it actually yields a different rotation because the angles are being rotated about the wrong axes now. Three.js [uses different formulas for the quaternion](https://github.com/mrdoob/three.js/blob/master/src/math/Quaternion.js#L214-L254) when applying euler angles in different orders.

Reply:
> Just swapping the order in which angles are passed into the function does not change the rotation order

Yep, I see my mistake now. Changed to the new formula.

> Three.js supports changing rotation order and uses different formulas for the quaternion when applying euler angles in different orders.

Yep, looks like they're using intrinsic ordering labels too.